### DISCUSSION TITLE
Unexpected style includes in production bundle

### DISCUSSION THREAD

**Discussion by adixoo** [tailwindlabs/tailwindcss#15019] [Help] [answered]
I'm facing an issue with the `@layer utilities` and `@layer components` directives. When I define custom classes within these layers, they seem to be included in the production bundle regardless of whether they're actually used in the project. 

If I define a `.button` class in either the `utilities` or `components` layer, it's bundled into the final build, even if I don't explicitly use it anywhere in my code. This leads to unnecessary increase of bundle size. 

Project link: [adixoo/tailwind-bug-or-feature](https://github.com/adixoo/tailwind-bug-or-feature)

I'm unsure if this is intended behavior or a bug. Should I report this as an issue or consider it a feature?

**Comment by wongjn** [comment, marked as answer]
This is a side effect of [how Tailwind scans files for classes](https://tailwindcss.com/docs/content-configuration#class-detection-in-depth). You have `<button className="i-will…` in `src/App.tsx`. Tailwind matches the `button` of this line with your `button` rule, thus the class is in the final build.

**Reply by adixoo** [reply]
Is there any way to fix this behaviour.

**Reply by wongjn** [reply]
You could:
- Remove the rule from the CSS file.
- Add the class name to the [`blocklist`](https://tailwindcss.com/docs/content-configuration#discarding-classes).
- Rename your class to something else less likely to collide, for example `.btn`.

**Reply by adixoo** [reply]
Ok got it 👍
